### PR TITLE
Add strong consistency riak_kv integration dev docs

### PR DESCRIPTION
**Pull request by engelsanchez**
This page explains the integration points between the Riak KV store and
the riak_ensemble library used to implement strongly consistent
operations.  The library comes with its own developer documents, which
are referenced here.

The script docs/update_line_numbers.erl can be used to keep references
to functions in the docs in sync with the code. Just run it like this
after compiling riak_kv:

escript docs/update_line_numbers.erl ebin docs/*.md

When reviewing, look at the branch in Github to verify links and layout  https://github.com/basho/riak_kv/blob/feature/add-sc-docs/docs/Strong-Consistency.md

### CODE REVIEW

**Comment by andrewjstone** [comment]
Taking review of this. I know there was such a huge backlog of possible reviewers ;)

File: docs/Strong-Consistency.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+Ensembles are mapped to preference lists, the replication groups assigned to an object by consistent hashing. Ensemble peers use a K/V backend that encapsulates riak_kv vnodes (4).
+
+1. [riak_ensemble Repository](https://github.com/basho/riak_ensemble)
+2. [riak_ensemble developer documentation](https://github.com/basho/riak_ensemble/tree/develop/doc)
+3. [Integration with riak_core](https://github.com/basho/riak_core/wiki#strong-consistency)
+4. [riak_kv_ensemble_backend](../src/riak_kv_ensemble_backend.erl)
+
+# Managing Ensembles
+
+An ensemble is created for each primary preference list. Unlike eventually consistent Riak, fallback vnodes never participate in consistent operations.  For example, if there are buckets with a replication factor (n_val) of 2 and some with 3, two ensembles will be created for each partition of the ring.  For partition zero, for example, we will have {0, 2} and {0, 3}.
```
Comment (andrewjstone) [inline docs/Strong-Consistency.md]: I'd change the tuples to {kv, 0, 2} and {kv, 0, 3} to match the actual ensemble names, and maybe use n_val of 3 and 5 as examples instead (along with their corresponding ensemble names).

**Comment by andrewjstone** [comment]
@engelsanchez :+1: on this with my suggested change

**Comment by engelsanchez** [comment]
Thanks @andrewjstone! Comments addressed in 92ac71b

**Comment by engelsanchez** [comment]
@borshop merge

**Pull request merged**